### DISCUSSION TITLE
Click handler with .outside modifier should not be registered if the element is not shown with x-if

### DISCUSSION THREAD

**Discussion by hongquan** [alpinejs/alpine#3642] [7. Bugs (archived)]
Now, if we have:

```html
<div x-data='{open: false}'
  <button @click='open = true'>Show modal</button>
  <template x-if='open'>
    <div>
      <div id='modal' @click.outside='open = false'>Content</div>
    </div>
  </template>
</div>
```

Everytime I click on the `<button>` element, the handler of `@click.outside` on `#modal` is also triggered.

**Comment by Tim-Wils** [comment]
Try .stop on the button trigger:  `<button @click.stop='open = true'>Show modal</button>`

**Reply by hongquan** [reply]
This works, thank!

**Comment by ekwoka** [comment]
hmm, it shouldn't be registered, at least until the first time the if is true, and it should then remove itself automatically when if is false....

x-show has a built in correction for such cases to ensure click outsides don't rehide the same thing. Maybe you can use that for now?

**Reply by hongquan** [reply]
Yes, I changed to use `x-show`, but I still prefer to use `x-if`.
